src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderBooking.kt

```diff
@@ -113,6 +113,15 @@ data class OffenderBooking(
   @OneToMany(mappedBy = "id.offenderBooking", cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
   val incentives: MutableList<Incentive> = mutableListOf(),
 ) {
+  fun getNextSequence(): Long {
+    val i = incentives.stream().max(Comparator.comparing { (a, _) -> a.sequence })
```

Comment:
You could have used Kotlin semantics here `incentives.map(it.sequence).maxOrNull()?. let {it + 1}   ?: 1`  or something like that

Reply:
Im glad you mentioned that ! , i couldnt work out what the 'native' way forward was with this